### PR TITLE
[arp]: Add tests for dualtor failed neighbors

### LINKED ISSUE #6298: [test_gap]: Dual ToR Failed IPv6 Neighbors
**Description**

When IPv6 entry is in failed state in one T0 but resolved in another T0, it likely means mux is 'standby' in the 'failed' T0 but 'active' in the reachable T0. In standby state, ping6 or ndisc6 is not automatically resolving 'failed' entry. In such cases, if a mux state change is initiated, the new active still has the entry in 'failed' state and can take upto 300sec (current arp_update interval) for it to be resolved by new active ToR.

This can cause packet loss for IPv6 destinations. 

A test is needed to cover this scenario

The test should:

1. For the same IPv6 neighbor, create a failed entry on the standby ToR and a reachable entry on the active ToR.
2. Run `arp_update` on the standby ToR
3. Confirm that the neighbor becomes INCOMPLETE and remains incomplete for some time (<10 seconds?)
4. Run `arp_update` on the active ToR
5. Confirm that the neighbor becomes REACHABLE

A second test is needed for the following scenario:

1. For a new IPv6 neighbor, create a reachable entry on the active ToR only.
2. Run `arp_update` on the active ToR
3. Confirm that the IPv6 neighbor entry was added to the standby ToR and is STALE

For both test cases, `arp_responder` should be enabled on the PTF

### PR DESCRIPTION
**Pull request by theasianpianist**
Signed-off-by: Lawrence Lee <lawlee@microsoft.com>

### Description of PR

Summary:
Fixes #6298 

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
We have a test gap for some scenarios involving failed neighbor entries on dualtor as outlined in #6298 

#### How did you do it?
Add two new test cases:
- `test_arp_update_for_failed_standby_neighbor` which checks that the `arp_update` script is able to recover a FAILED neighbor  entry on the standby TOR to REACHABLE
- `test_standby_unsolicited_neigh_learning` which checks that the standby TOR is able to learn a new, REACHABLE neighbor entry triggered by `arp_update` on the active TOR

Add new fixtures:
- `config_dualtor_arp_responder` which configures the `arp_responder` service on the PTF to reply to ARP requests for the server IPs configured in the MUX_CABLE table on the TORs
- `pause_garp_service` to temporarily stop `garp_service` on the PTF for test cases where it would interefere

Cleanup some import statements

#### How did you verify/test it?
Run `test_arp_dualtor.py`, verify all test cases pass for both IPv4 and IPv6

#### Supported testbed topology if it's a new test case?
True dual ToR only

### CODE REVIEW

File: tests/common/dualtor/dual_tor_utils.py [hunk 1]
```diff
@@ -1410,3 +1413,27 @@ def is_tunnel_qos_remap_enabled(duthost):
...
+        arp_responder_conf[ptf_intf] = [
+            str(ipaddress.ip_interface(config_vals["SERVER"]["IPv4"]).ip),
+            str(ipaddress.ip_interface(config_vals["SERVER"]["IPv6"]).ip)]
+
+    ptfhost.copy(content=json.dumps(arp_responder_conf, indent=4, sort_keys=True), dest="/tmp/from_t1.json")
+    ptfhost.host.options["variable_manager"].extra_vars.update({"arp_responder_args": ""})
+    ptfhost.template(src="templates/arp_responder.conf.j2", dest="/etc/supervisor/conf.d/arp_responder.conf")
+
+    supervisor_cmd = "supervisorctl reread && supervisorctl update && supervisorctl restart arp_responder"
+    ptfhost.shell(supervisor_cmd)
```
Comment (lolyu) [inline tests/common/dualtor/dual_tor_utils.py:1439]: Do we need to stop the `arp_responder` after the testcases?

**Review by lolyu — DISMISSED** [review]

File: tests/common/dualtor/dual_tor_utils.py (hunk 1 above)
Comment (theasianpianist) [reply tests/common/dualtor/dual_tor_utils.py:1439]: Good call, pushed a new commit to stop it during teardown

**Comment by wangxin** [comment]
@theasianpianist There are many conflicts while merging to 202012 branch. Can you create a separate PR to 202012 branch to have this change?

**Comment by theasianpianist** [comment]
> @theasianpianist There are many conflicts while merging to 202012 branch. Can you create a separate PR to 202012 branch to have this change?

PR here, please check https://github.com/sonic-net/sonic-mgmt/pull/6878

**Pull request merged**